skylib/digester.bzl

```
@@ -0,0 +1,73 @@
+load(
+    "@io_bazel_rules_docker//container:layer_tools.bzl",
+    _gen_img_args = "generate_args_for_image",
+    _get_layers = "get_from_target",
+)
+
+def calc_digest(ctx):
+    """ Calculate the digest for a given image
+
+    Args:
+       ctx: The context
+    Returns:
+        image: The image which has reference to a digest
+    """
+
+    digester_args = []
+    digester_input = []
+
+    image = _get_layers(ctx, ctx.label.name, ctx.attr.image)
+    digester_img_args, digester_img_inputs = _gen_img_args(ctx, image)
+    digester_input += digester_img_inputs
+    digester_args += digester_img_args
+
+    digester_args += ["--dst", str(ctx.outputs.digest.path), "--format", str(ctx.attr.format)]
+
+    ctx.actions.run(
+        inputs = digester_input,
+        outputs = [ctx.outputs.digest],
+        executable = ctx.executable._digester,
+        arguments = digester_args,
+        tools = ctx.attr._digester[DefaultInfo].default_runfiles.files,
+        mnemonic = "CalculateDigest",
+    )
+
+    return image
+
+def _calculate_digest_impl(ctx):
+    calc_digest(ctx)
+
+# Rule definition
+calculate_digest = rule(
```

Avoid using verbs in rule names. targets support to describe _what_ in the output, not _how_ output is created.
```suggestion
image_digest = rule(
```